src/clp_ffi_js/ir/StructuredIrUnitHandler.cpp

```diff
@@ -0,0 +1,145 @@
+#include "StructuredIrUnitHandler.hpp"
+
+#include <algorithm>
+#include <cctype>
+#include <cstddef>
+#include <iterator>
+#include <memory>
+#include <string>
+#include <string_view>
+#include <type_utils.hpp>
+#include <utility>
+
+#include <clp/ffi/ir_stream/decoding_methods.hpp>
+#include <clp/ffi/KeyValuePairLogEvent.hpp>
+#include <clp/ffi/SchemaTree.hpp>
+#include <clp/ffi/Value.hpp>
+#include <emscripten/val.h>
+#include <spdlog/spdlog.h>
+
+#include <clp_ffi_js/constants.hpp>
+#include <clp_ffi_js/ir/LogEventWithFilterData.hpp>
+
+namespace clp_ffi_js::ir {
+
+namespace {
+/**
+ * Parses the `LogLevel` from an input string.
+ * @param str
+ * @return
+ */
+auto parse_log_level(std::string_view str) -> LogLevel;
+
+auto parse_log_level(std::string_view str) -> LogLevel {
+    LogLevel log_level{LogLevel::NONE};
+
+    // Convert the string to uppercase.
+    std::string log_level_name_upper_case{str};
+    std::ranges::transform(
+            log_level_name_upper_case.begin(),
+            log_level_name_upper_case.end(),
+            log_level_name_upper_case.begin(),
+            [](unsigned char c) { return std::toupper(c); }
+    );
```

Comment:
It is good to handle where the log levels are stored fully or partially in lower cases but this is uncommon. Shall we consider removing this for now and add the handling when we implement the new format string syntax?

Reply:
> Also how is format string re: to this?

Supposedly with the new format string syntax, we can configure how to interpret the raw log event fields.

Reply:
Right, you got me. i was fuzzily picturing that we could transform the log level during some preprocessing stage of the new formatter, but now with the new syntax implemented i agree that idea doesn't seem to easily / cleanly fit anywhere. Let's keep this